src/fsharp/service/IncrementalBuild.fs

```diff
@@ -1211,6 +1207,17 @@ type IncrementalBuilder(tcGlobals,
         | Some(boundModel, timestamp) -> PartialCheckResults(boundModel, timestamp) |> Some
         | _ -> None
 
+    member builder.TryGetCheckResultsForFileInProject (filename) =
+        let cache = TimeStampCache defaultTimeStamp
+        let state = currentState
+        let state = computeStampedFileNames state cache
+        let state = computeStampedReferencedAssemblies state cache
```

Comment:
are these refactorable to take `state` as the last param so you can do pipelines? More of a style thing but any time I see shadowing like this it feels a little like a code smell

Reply:
There will be more cleanup to do in this area. I know @dsyme has ideas, but they will be addressed in a separate PR.